rnd/autogpt_builder/src/components/ui/RunnerUI/RunnerInputUI.tsx

```diff
@@ -0,0 +1,165 @@
+                blockInputs.map((block) => (
+                  <div key={block.id} className="space-y-1">
+                    <h3 className="text-base font-semibold">
+                      {block.hardcodedValues.name || "Unnamed Input"}
+                    </h3>
+
+                    {block.hardcodedValues.description && (
+                      <p className="text-sm text-gray-600">
+                        {block.hardcodedValues.description}
+                      </p>
+                    )}
+
+                    <div>
+                      {block.hardcodedValues.placeholder_values &&
+                      block.hardcodedValues.placeholder_values.length > 1 ? (
+                        <Select
+                          onValueChange={(value) =>
+                            onInputChange(block.id, "value", value)
+                          }
+                          value={block.hardcodedValues.value?.toString() || ""}
+                        >
+                          <SelectTrigger className="w-full">
+                            <SelectValue placeholder="Select a value" />
+                          </SelectTrigger>
+                          <SelectContent>
+                            {block.hardcodedValues.placeholder_values.map(
+                              (placeholder, index) => (
+                                <SelectItem
+                                  key={index}
+                                  value={placeholder.toString()}
+                                >
+                                  {placeholder.toString()}
+                                </SelectItem>
+                              ),
+                            )}
+                          </SelectContent>
+                        </Select>
+                      ) : (
+                        <Input
+                          id={`${block.id}-Value`}
+                          value={block.hardcodedValues.value?.toString() || ""}
+                          onChange={(e) =>
+                            onInputChange(block.id, "value", e.target.value)
+                          }
+                          placeholder={
+                            block.hardcodedValues.placeholder_values?.[0]?.toString() ||
+                            "Enter value"
+                          }
+                          className="w-full"
+                        />
+                      )}
+                    </div>
+                  </div>
+                ))
+              ) : (
+                <p>No input blocks available.</p>
+              )}
+            </div>
+          </ScrollArea>
+        </div>
+        <DialogFooter className="px-6 py-4">
+          <Button
+            onClick={handleRun}
+            className="px-8 py-2 text-lg"
+            disabled={isRunning}
+          >
+            {isRunning ? "Running..." : "Run"}
+          </Button>
+        </DialogFooter>
+      </DialogContent>
+    </Dialog>
+  );
```

Comment:
Can we group parts of this and make smaller components out of them? imo 100-200 lines is where I consider breaking out a bit of it into sub components for readability